File: skills/code-security-audit-v6/tests/fixtures/ts-express/src/db/query.ts
import { Pool } from 'pg';
const pool = new Pool();

export async function executeQuery(sql: string, params: any[]) {
  const result = await pool.query(sql, params);
  return result.rows;
}

export async function rawQuery(userInput: string) {
  const result = await pool.query(`SELECT * FROM users WHERE name = '${userInput}'`);
  return result.rows;
}

File: skills/code-security-audit-v6/tests/fixtures/go-gin/db/query.go
package db

import (
	"database/sql"
	"fmt"
)

var DB *sql.DB

func ExecuteQuery(query string, args ...interface{}) (*sql.Rows, error) {
	return DB.Query(query, args...)
}

func UnsafeQuery(userInput string) (sql.Result, error) {
	query := fmt.Sprintf("SELECT * FROM users WHERE name = '%s'", userInput)
	return DB.Exec(query)
}
